File: practice/convert/example2_book_library/book-library/src/main/java/dev/starryeye/book_library/controller/user/RegisterUserController.java
package dev.starryeye.book_library.controller.user;

import dev.starryeye.book_library.application.command.result.RegisterUserResult;
import dev.starryeye.book_library.controller.user.request.RegisterUserRequest;
import dev.starryeye.book_library.controller.user.response.UserResponse;
import dev.starryeye.book_library.application.command.RegisterUserService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class RegisterUserController {

    private final RegisterUserService service;

    @PostMapping("/api/v1/users/new")
    public ResponseEntity<UserResponse> registerUser(@Valid @RequestBody RegisterUserRequest request) {
        RegisterUserResult result = service.register(request.toCommand());
        UserResponse response = UserResponse.of(result);
        return ResponseEntity.ok(response);
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/controller/user/RegisterUserController.kt
package dev.starryeye.book_library.controller.user

import dev.starryeye.book_library.application.user.command.RegisterUserService
import dev.starryeye.book_library.controller.user.request.RegisterUserRequest
import dev.starryeye.book_library.controller.user.request.toCommand
import dev.starryeye.book_library.controller.user.response.UserResponse
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class RegisterUserController(
    private val service: RegisterUserService
) {

    @PostMapping("/api/v1/users/new")
    fun registerUser(
        @Valid @RequestBody request: RegisterUserRequest
    ): UserResponse {
        val result = service.register(request.toCommand())
        return UserResponse.of(result)
    }
}